scripts/Deploy-CustomScriptsExtension-v2.ps1

```diff
@@ -150,6 +150,16 @@ else
 # the following keys: name, value, valueType (optional: defaults to string) and their accompanying values
 # the only supported valueTypes are: string (default) and FilePath
 # for FilePath valueType, the path will be tested, content read and base64 encoded before being passed as the parameter value
+
+# we already have some userful parameters that downstream scripts can leverage
+Log-Message "Adding Azure Subscription and AAD related parameters"
+
+$upgradeParameters += @{"name"="azure-resource-group"; "value"=$ResourceGroupName}
```

Comment:
You'd need to clarify the question. Do you mean why are we using the powershell script?
 scripts/Deploy-CustomScriptsExtension-v2.ps1 to deploy?